NeuroID/src/test/java/com/neuroid/tracker/SensorHelperTests.kt

```diff
@@ -0,0 +1,152 @@
+package com.neuroid.tracker
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorManager
+import androidx.core.content.getSystemService
+import com.neuroid.tracker.callbacks.NIDSensorData
+import com.neuroid.tracker.callbacks.NIDSensorGenListener
+import com.neuroid.tracker.callbacks.NIDSensorHelper
+import com.neuroid.tracker.callbacks.NIDSensorStatus
+import com.neuroid.tracker.callbacks.NIDSensors
+import com.neuroid.tracker.utils.NIDLogWrapper
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.Before
+import org.junit.Test
+import java.lang.reflect.Field
+
+class SensorHelperTests {
+    @Before
+    fun setup() {
+        NIDSensorHelper.resetSensorManager()
+    }
+
+    @Test
+    fun initSensorHelper_gyro_test() {
+        val nidSensors = mockk<NIDSensors>()
+        val logger = mockk<NIDLogWrapper>()
+        every { logger.i(any(  ), any()) } just runs
+        val gyro = mockk<Sensor>()
+        every { gyro.type } returns Sensor.TYPE_GYROSCOPE
+        every { gyro.name } returns "gyro"
+        val sensorManager = mockk<SensorManager>()
+        every { sensorManager.getSensorList(Sensor.TYPE_ALL) } returns listOf(gyro)
+        every { sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE) } returns gyro
+        every { sensorManager.registerListener(any<NIDSensorGenListener>(), any(), any(), any<Int>())} returns true
+        every { sensorManager.requestTriggerSensor(any(), any())} returns true
+        val context = mockk<Context>()
+        every { context.getSystemService<SensorManager>() } returns sensorManager
+        NIDSensorHelper.initSensorHelper(context, logger, nidSensors)
+        // is it the proper sensor?
+        verify { logger.i("NIDSensorHelper", "Sensor:gyro 4") }
+        // did we register the listener?
+        verify { sensorManager.registerListener(any(), gyro, 10000, 10000) }
```

Comment:
NIT: Could we consider breaking some of these into reusable functions that the other test also use to make it more readable?

I.e. useMockedLogger() - similar to how its done for NeuroIDUnitTests?
I am happy to go through and refactor if you'd like

Reply:
Sure can break this down a bit. Thanks for volunteering 👍